girder/api/rest.py

```diff
@@ -298,9 +298,10 @@ def endpointDecorator(self, *args, **kwargs):
             if 'Content-Range' in cherrypy.response.headers:
                 cherrypy.response.status = 206
 
-            if isinstance(val, types.FunctionType):
-                # If the endpoint returned a function, we assume it's a
-                # generator function for a streaming response.
+            if callable(val):
+                # If the endpoint returned anything callable (function,
+                # lambda, functools.partial), we assume it's a generator
+                # function for a streaming response.
```

Comment:
I don't think so. A `cherrypy.lib.file_generator` is not callable, as it does not contain an `__call__` method (not even one implicitly defined) and thus raises `TypeError: 'file_generator' object is not callable` if you try to call it. Feel free to try
```
import cherrypy
from cStringIO import StringIO
callable(cherrypy.lib.file_generator(StringIO()))
```
if you'd like.

I'm sure @zachmullen knows this next part, but I'll elaborate for the benefit of others: What may be causing some confusion is the particular way that Girder returns streaming responses. Rather than just returning a generator iterator for streaming responses, and testing whether it's a `types.GeneratorType` in this block, Girder typically returns the whole generator function. This function must then be trivially called once, whereby it returns a pristine generator iterator. `cherrypy.lib.file_generator` is an attempt to emulate this sort of generator iterator, not the generator function.

The only time you need the function itself is if you plan on iterating through it multiple times. Since that's not the case, Girder could have been designed to just have streaming responses be returned as generator iterators (or really any type of iterator). On the other hand, requiring streaming responses to return the original function is a nice protection against developers actually returning an iterator instead of a data structure, and accidentally producing a streaming response instead of getting automatic JSON-encoding.